File: acme-admin/src/main/java/com/acme/admin/controller/RevenueController.java
package com.acme.admin.controller;

import com.acme.admin.domain.RevenueEntity;
import com.acme.admin.dto.RevenueCreateRequest;
import com.acme.admin.service.RevenueService;
import jakarta.validation.Valid;
import java.util.List;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/finance/revenue")
public class RevenueController {
  private final RevenueService service;

  public RevenueController(RevenueService service) {
    this.service = service;
  }

  @GetMapping
  public List<RevenueEntity> list() {
    return service.list();
  }

  @PostMapping
  public RevenueEntity create(@Valid @RequestBody RevenueCreateRequest req) {
    return service.create(req);
  }
}

File: acme-admin-dotnet/Controllers/RevenueController.cs
using Acme.Admin.Api.Domain;
using Acme.Admin.Api.DTO;
using Acme.Admin.Api.Security;
using Acme.Admin.Api.Services;
using Acme.Admin.Api.Validation;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Acme.Admin.Api.Controllers;

[ApiController]
[Route("api/finance/revenue")]
public sealed class RevenueController(RevenueService service) : ControllerBase
{
    [HttpGet]
    [Authorize(Policy = PermissionPolicies.FinanceRead)]
    public async Task<ActionResult<PagedResult<RevenueEntity>>> List(
        [FromQuery] int page = Paging.DefaultPage,
        [FromQuery] int size = Paging.DefaultSize,
        CancellationToken ct = default)
    {
        var (p, s) = Paging.Normalize(page, size);
        return Ok(await service.ListAsync(p, s, ct));
    }

    [HttpPost]
    [Authorize(Policy = PermissionPolicies.FinanceWrite)]
    public async Task<ActionResult<RevenueEntity>> Create([FromBody] RevenueCreateRequest req, CancellationToken ct)
    {
        return Ok(await service.CreateAsync(req, ct));
    }
}
